tests/foreman/api/test_host.py

```diff
@@ -903,3 +910,151 @@ def test_negative_update_os(self):
             host = host.update(['operatingsystem'])
         self.assertNotEqual(
             host.read().operatingsystem.read().name, new_os.name)
+
+    @run_only_on('sat')
+    @stubbed
+    @tier3
+    def test_positive_create_baremetal_bios(self):
+        """Create a new Host from provided MAC address
+
+        @id: 9d74ed70-3197-4825-bf96-21eeb4a765f9
+
+        @steps:
+        1. Create a PXE-based VM with BIOS boot mode (outside of Satellite).
+        2. Create a new host using 'BareMetal' option and MAC address of the
+        pre-created VM
+
+        @assert: Host is created
+
+        @caseautomation: System
+
+        """
+
+    @run_only_on('sat')
+    @stubbed
+    @tier3
+    def test_positive_create_baremetal_uefi(self):
+        """Create a new Host from provided MAC address
+
+        @id: 9b852c4d-a94f-4ba9-b666-ea4718320a42
+
+        @steps:
+        1. Create a PXE-based VM with UEFI boot mode (outside of Satellite).
+        2. Create a new host using 'BareMetal' option and MAC address of the
+        pre-created VM
+
+        @assert: Host is created
+
+        @caseautomation: System
+
+        """
+
+    @run_only_on('sat')
+    @stubbed
+    @tier3
+    def test_positive_verify_tftp_files_pxegrub_uefi(self):
+        """Provision a new Host and verify the tftp file structure is correct
+
+        @id: 0c51c8ad-858c-44e1-8b14-8e0c52c29da1
+
+        @steps:
+        1. Associate a pxegrub-type provisioning template with the os
+        2. Create new host (can be fictive bare metal) with the above OS and
+        PXE loader set to Grub UEFI
+        3. Build the host
+
+        @assert:
+        Verify [/var/lib/tftpboot/] contains the following dir/file structure:
```

Comment:
the main difference is in the grub vs grub2 dirs, in the filename format (the format of the MAC address, etc.). However, i have this wrong, there are duplicit entries caused by copypasta. Will fix it.